keep/providers/splunk_provider/splunk_provider.py

```diff
@@ -0,0 +1,154 @@
+        }
+    )
+    password: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "Splunk Password",
+            "sensitive": True,
+        }
+    )
+
+    host: str = dataclasses.field(
+        metadata={
+            "description": "Splunk Host (default is localhost)",
+        },
+        default="localhost"
+    )
+    port: int = dataclasses.field(
+        metadata={
+            "description": "Splunkd Port (default is 8089)",
+        },
+        default=8089
+    )
+
+
+class SplunkProvider(BaseProvider):
+    """Pull alerts and query incidents from Splunk."""
+
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="connect_to_client",
+            description="The user can connect to the client",
+            mandatory=True,
+            alias="Connect to the client",
+        )
+    ]
+
+    SEVERITIES_MAP = {
+        "1": AlertSeverity.LOW,
+        "2": AlertSeverity.INFO,
+        "3": AlertSeverity.WARNING,
+        "4": AlertSeverity.HIGH,
+        "5": AlertSeverity.CRITICAL,
+    }
+
+    def __init__(
+            self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        super().__init__(context_manager, provider_id, config)
+
+    def validate_scopes(self) -> dict[str, bool | str]:
+        try:
+            connect(username=self.authentication_config.username, password=self.authentication_config.password,
+                    host=self.authentication_config.host, port=self.authentication_config.port)
+            scopes = {
+                "connect_to_client": True,
+            }
+        except Exception as e:
+            self.logger.exception("Error validating scopes")
+            scopes = {
+                "connect_to_client": str(e),
+            }
+        return scopes
+
+    def validate_config(self):
+        self.authentication_config = SplunkProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def dispose(self):
+        """
+        No need to dispose of anything, so just do nothing.
+        """
+        pass
+
+    def setup_webhook(
+            self, tenant_id: str, keep_api_url: str, api_key: str, setup_alerts: bool = True
+    ):
+        self.logger.info("Setting up Splunk webhook on all Alerts")
+        creation_updation_kwargs = {
+            "actions": "webhook",
+            "action.webhook": "1",
+            "action.webhook.param.url": keep_api_url,
+        }
+        service = connect(username=self.authentication_config.username, password=self.authentication_config.password,
+                          host=self.authentication_config.host, port=self.authentication_config.port)
+        for saved_search in service.saved_searches:
+            existing_webhook_url = saved_search["_state"]["content"].get("action.webhook.param.url", None)
+            if existing_webhook_url is None or existing_webhook_url != keep_api_url:
```

Comment:
So a saved search can't have more than 1 webhook action? If the user already defined some webhook action for that saved search, he won't get that in Keep? is there a way to add multiple?

Reply:
No, I don't see any such option, if there is a webhook is added, you cannot add more webhooks
![image](https://github.com/keephq/keep/assets/70096901/5d379f8a-ee0f-4798-91e1-a312257160bc)
![image](https://github.com/keephq/keep/assets/70096901/3ebcfad8-9547-492e-aea4-0840ce87a813)

Reply:
![CleanShot 2024-03-12 at 12 16 40](https://github.com/keephq/keep/assets/68807791/6d3d148f-62cd-4b72-9fd9-09c9cf2a09db)
doesn't let you add webhook twice?

Reply:
@tal, no, once added it is removed from the add action button, that's what is shown in the second screenshot

Reply:
Also in the API response, there is only one field for webhook URL, so that is an additional verification that only one webhook is allowed